### DISCUSSION TITLE
Can I use Thanos Query Front to Proxy Prometheus

### DISCUSSION THREAD

**Discussion by zxhdaniel** [thanos-io/thanos#5774] [Questions & Answers] [answered]
In my company, My upstream department writes data to influxdb. I want to have a global read view exposed to Grafana.
Metrics data is stored in influxdb but not oss. Since Prometheus supports multiple remote readers,  I think it can replace Thanos Query. Furthermore, I want to use Thanos Query Front to speed up my query to Prometheus. Finally I got this architecture. I have tested this architecture.  Is it any problem? I worry about that I may miss something.
![image](https://user-images.githubusercontent.com/2572041/194735583-642dadfa-6235-4e30-b106-2e094e9167ad.png)

**Comment by yeya24** [comment, marked as answer]
For the question from the discussion title, yes you can use QF to proxy prometheus.
